node/src/storage_config.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Storage configuration
+
+use serde::{Deserialize, Serialize};
+
+/// Storage type to use
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub enum StorageBackend {
+    Lmdb,
+    InMemory,
+}
+
+impl std::str::FromStr for StorageBackend {
+    type Err = StorageBackendParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "lmdb" => Ok(StorageBackend::Lmdb),
+            "inmemory" | "in-memory" => Ok(StorageBackend::InMemory),
+            name => {
+                let name = name.into();
+                Err(StorageBackendParseError::Unrecognized { name })
+            }
+        }
+    }
+}
+
+#[derive(thiserror::Error, Debug, PartialEq, Eq)]
+pub enum StorageBackendParseError {
+    #[error("Unrecognized backend '{name}'")]
+    Unrecognized { name: String },
+}
+
+/// Storage configuration
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct StorageConfig {
```

Comment:
I don't understand why we need a separate storage config. Storage is part of chainstate. Two things related should be there: The storage type, and the data directory. The relation between lmdb's storage path and the data directory is fixed. It's only a subdirectory of that, say, `chainstate-lmdb`. I still haven't dug deep in the PR to see how it's done here, but seeing a separate storage configuration seems redundant.